pkg/wallet/wallet.go

```diff
@@ -61,7 +68,7 @@ func (w *BaseWallet) CreateEncryptionKey() (string, error) {
 	}
 	base58Pub := base58.Encode(pub[:])
 	// TODO - need to encrypt the priv before putting them in the store.
-	if err := w.persistKey(base58Pub, &cryptoutil.KeyPair{Pub: pub[:], Priv: priv[:]}); err != nil {
+	if err := persist(w.keystore, base58Pub, &cryptoutil.KeyPair{Pub: pub[:], Priv: priv[:]}); err != nil {
```

Comment:
this function should be part of the store...

the call should be `w.keystore.Persist()` ..

Reply:
Moved logic to common private function since this logic (marshal + check error + put) was repeated at many places in wallet.